src/components/StartPrjModal/CreatePrjModal/index.tsx

```diff
@@ -1,30 +1,490 @@
-import { FC } from 'react';
-import BaseModal from 'components/Home/BaseModal';
+import { FC, useEffect, useState } from 'react';
+import Modal from 'components/Modal';
 import styles from './index.module.scss';
-
+import Select from 'react-select';
+import timezoneData from 'assets/data/timezones.json';
+import { customStyles } from './selectStyles';
+import getRandomString from 'utils/getRandomString';
+import toast, { Toaster } from 'react-hot-toast';
+import { useMutation } from 'react-query';
+import config from 'config';
+import { useNavigate } from 'react-router-dom';
 interface ICreatePrjProps {
   onNext: () => void;
   onClose: () => void;
+  orgId: number;
 }
 
-const CreatePrjModal: FC<ICreatePrjProps> = ({ onClose, onNext }) => {
+const CreatePrjModal: FC<ICreatePrjProps> = ({ onClose, onNext, orgId }) => {
+  const navigate = useNavigate();
+  const temp: any = [];
+  useEffect(() => {
+    if (!temp.length) {
+      timezoneData.forEach((element) => {
+        temp.push({ value: element.value, label: element.text });
+      });
+      setOptions(temp);
+    }
+  }, []);
+
+  const [formData, setFormData] = useState<any>({
+    name: '',
+    description: '',
+    timeOfCompletion: '',
+    preferredTimeZones: [],
+    organisationId: '',
+    flResources: [],
+    flProjectCategory: [{ categoryName: '', percentageAllocation: '' }],
+  });
+  const [error, setError] = useState<any>({
+    message: '',
+  });
+  const [options, setOptions] = useState<any>(null);
+  const [selectedOptions, setSelectedOptions] = useState<any>(null);
+  const [addMoreResources, setAddMoreResources] = useState<any>({ counter: 0 });
+  const [addMoreCategories, setAddMoreCategories] = useState<any>({
+    counter: 0,
+  });
+  const [submit, setSubmit] = useState<boolean>(false);
+  const mutation = useMutation('addProject', () =>
+    fetch(`${config.ApiBaseUrl}/fl-project`, {
+      method: 'POST',
+    }).then((response) => response.json())
+  );
+  useEffect(() => {
+    setFormData((prevState: any) => {
+      const selectedOptionsLabel = selectedOptions?.map((option: any) => {
+        return option.label;
+      });
+      // if selectedOptionslabel has length of 1 or more
+      if (selectedOptionsLabel?.length === 1)
+        prevState.preferredTimeZones = selectedOptionsLabel[0];
+      prevState.preferredTimeZones = selectedOptionsLabel?.join(', ');
+      return {
+        ...prevState,
+      };
+    });
+  }, [selectedOptions]);
+
+  useEffect(() => {
+    if (submit) {
+      console.log(formData);
+      const json: any = JSON.stringify(formData);
+      console.log(json);
+      // mutation.mutate(json);
+      navigate('/project');
+    }
+
+    if (!formData.organisationId) {
+      setFormData((prevState: any) => {
+        prevState.organisationId = orgId;
+        return {
+          ...prevState,
+        };
+      });
+    }
+    if (formData.name === '' || formData.description === '') {
+      setError((prevState: any) => ({
```

Comment:
For validation, you can check inside `validations` folder how I've separated it out in a different file.